spec/factories/samples.rb

```diff
@@ -23,13 +23,11 @@
 
     # metadata fields
     # ensure the metadata field is added to the host genome
-    # TODO(tiago): will not work if the host genome was created before.
-    # COuld find out why the host genome metadata fields could not be updated.
-    before(:create) do |_sample, options|
+    before(:create) do |sample, options|
       options.metadata_fields.each_key do |metadata_field_name|
-        unless MetadataField.exists?(name: metadata_field_name)
-          create(:metadata_field, name: metadata_field_name, default_for_new_host_genome: 1)
-        end
+        metadata_field = create(:metadata_field, name: metadata_field_name, default_for_new_host_genome: 1)
+        # This is needed to make metadata pass validation below
+        sample.host_genome.metadata_fields << metadata_field
```

Comment:
I reviewed this briefly with @MarkAZhang

Reply:
I don't see why it should... but then the test is failing now with a single error related to metadata: 

![image](https://user-images.githubusercontent.com/28797/61673142-2b714100-aca3-11e9-9713-e7d7e32ffc46.png)

Any ideas @MarkAZhang ?

Reply:
The test passes for me locally.

Reply:
Because it will create a the same metadata field twice. That should fail if we have a restriction on unique names. Thats why the unless is there. Do you mind trying to associate the same metadata field with two samples?
BTW, the above error does not seem related to the this particular potential problem.

Reply:
@tfrcarvalho I see now. I took out the `unless` for testing and I forgot to put it back. Now fixed. Good catch!